google-cloud-storage/src/main/java/com/google/cloud/storage/BlobInfo.java

```diff
@@ -260,6 +261,9 @@ static CustomerEncryption fromPb(StorageObject.CustomerEncryption customerEncryp
      */
     public abstract Builder setCrc32c(String crc32c);
 
+    /** Sets the custom time for an object. */
+    public abstract Builder setCustomTime(Long customTime);
```

Comment:
Why is this abstract? If it's concrete it wouldn't be a breaking change.

Reply:
It's the development model we've used over the years. It was designed before my time and folks who worked on the library have since left the team.

We could update it to concrete method instead but we would start changing development flow. Maybe that's a good thing, but I'd prefer to prevent changing development flow until we have a new design that fits into that model.

Reply:
The pot roast principle applies:

https://www.psychologytoday.com/us/blog/thinking-makes-it-so/201402/the-pot-roast-principle

Reply:
If it's implemented, it could be implemented to raise an exception of it not being implemented. Seems a little clunky, but would remove the need to create a breaking change.

Thanks for raising the simple questions @elharo, I haven't applied this train of thought to this problem, and I appreciate the feedback.

Reply:
@athakor please implement this method instead of using abstract to prevent the breaking change. Instead raise an exception of unimplemented if called directly.

Reply:
done